### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by pchaussalet**
Adds fields validation to inspectors.
To be validated, an inspector has to inherit from `AbstractInspector` and to have an instance of `ValidationController` in its serialization, bound both to `@owner.validationController` and to `inspector.validationController`.
The ValidationController uses the blueprints to match inspector's components name and object's property name. It is also possible to link a property to some specific fields when name doesn't match by using `validationController.fields` property (it is an object with property names as keys and component(s) as values).

### CODE REVIEW

File: core/service/validation-service.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+var Montage = require("montage").Montage,
+    Services = require("core/model/services").Services,
+    FastMap = require("collections/fast-map"),
+    FastSet = require("collections/fast-set");
+
+exports.ValidationService = Montage.specialize({
+    ACTIONS: {
```
Comment (thibaultzanini) [inline core/service/validation-service.js]: I know there is just one validation service within the app but that could be set on the constructor :)

File: ui/abstract/abstract-inspector.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    superEnterDocument: {
+        value: function() {
+            this._callSuperMethod('enterDocument', arguments);
+            AbstractInspector.prototype._abstractInspectorenterDocument.call(this);
+        }
+    },
+
+    superExitDocument: {
+        value: function() {
+            this._callSuperMethod('exitDocument');
```
Comment (thibaultzanini) [inline ui/abstract/abstract-inspector.js]: our super :(

File: ui/controls/cascading-list.reel/cascading-list.js [hunk 3]
```diff
@@ -151,11 +152,16 @@ exports.CascadingList = Component.specialize({
...
                 columnIndex = Math.min(self._stack.length, columnIndex);
-                self._push({
+                var context = {
                     object: object,
                     userInterfaceDescriptor: userInterfaceDescriptor,
                     columnIndex: columnIndex
-                });
+                };
+                if (self._defaultSelection && self._defaultSelection.error && self._defaultSelection.path.length == 0) {
+                    context.error = self._defaultSelection.error;
```
Comment (thibaultzanini) [inline ui/controls/cascading-list.reel/cascading-list.js:161]: Is `path` always defined?

File: ui/inspectors/calendar-task.reel/calendar-task.js [hunk 4]
```diff
@@ -36,7 +36,8 @@ exports.CalendarTask = AbstractComponentActionDelegate.specialize({
 
     exitDocument: {
         value: function() {
-            AbstractComponentActionDelegate.prototype.exitDocument.call(this);
+            this.superExitDocument();
+            this.superExitDocument();
```
Comment (thibaultzanini) [inline ui/inspectors/calendar-task.reel/calendar-task.js]: twice?

File: ui/inspectors/service.reel/service.js [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
  * @class Service
  * @extends Component
  */
-exports.Service = Component.specialize({
+exports.Service = AbstractInspector.specialize({
     systemGeneral: {
         value: null
     },
 
+    exitDocument: {
```
Comment (thibaultzanini) [inline ui/inspectors/service.reel/service.js]: Maybe it should be added on the AbsctratInspector

File: ui/inspectors/user.reel/user.js [hunk 6]
```diff
@@ -118,7 +119,8 @@ exports.User = AbstractComponentActionDelegate.specialize({
 
     exitDocument: {
         value: function() {
-            AbstractComponentActionDelegate.prototype.exitDocument.call(this);
+            this.superExitDocument();
```
Comment (thibaultzanini) [inline ui/inspectors/user.reel/user.js]: twice?

File: ui/inspectors/virtual-machine-volume.reel/virtual-machine-volume.js [hunk 7]
```diff
@@ -17,8 +17,15 @@ exports.VirtualMachineVolume = Component.specialize({
         }
     },
 
+    exitDocument: {
```
Comment (thibaultzanini) [inline ui/inspectors/virtual-machine-volume.reel/virtual-machine-volume.js]: same here

**Comment by Fzzr** [comment]
CRUD works normally in Volumes, Shares, Datasets, Snapshots, Users, Groups, Network Interfaces, and VMs. 

All services work as expected.

Editing the network configuration results in this non-fatal error at the command line: 

`:8083/ui/inspectors/inspector.reel/inspector.js:153: Uncaught TypeError: Cannot read property '_isNew' of undefined`.

This is 100% reproducible, but all GUI and middleware behavior is otherwise as normal.

File: ui/controls/cascading-list.reel/cascading-list.js (hunk 3 above)
Comment (pchaussalet) [reply ui/controls/cascading-list.reel/cascading-list.js:161]: Yes, and it wouldn't make any sense to have a selection without a path.

File: ui/inspectors/calendar-task.reel/calendar-task.js (hunk 4 above)
Comment (pchaussalet) [reply ui/inspectors/calendar-task.reel/calendar-task.js]: Damn, I thought I fixed that... Will do.

File: ui/inspectors/user.reel/user.js (hunk 6 above)
Comment (pchaussalet) [reply ui/inspectors/user.reel/user.js]: Damn, I thought I fixed that... Will do.

**Comment by thibaultzanini** [comment]
LGTM

**Pull request merged**